### DISCUSSION TITLE
"Raw" or "No-role" mode to simply call GPT with your prompt and nothing else?

### DISCUSSION THREAD

**Discussion by giladbarnea** [TheR1D/shell_gpt#551] [Ideas]
Sometimes, I just want to use GPT in my terminal, because I prefer to stay in the terminal over switching to a browser.

In other words, I don't always want a GPT that's a programming and system administration assistant which responds in 100 words or less (which is the default role), or any other specialized GPT. I just want "the" GPT, like OpenAI designed it.

I was thinking about adding a `--raw` option for that case.

What do you guys think? Would you find that useful?

I've implemented this for personal use, the code is available [here](https://github.com/giladbarnea/shell_gpt/tree/raw-mode).

![image](https://github.com/TheR1D/shell_gpt/assets/18548473/9ef28f95-9919-43ad-8293-69d2888b8b7d)

**Comment by giladbarnea** [comment]
I've prepared a pull request, in case the authors want to incorporate the feature:
https://github.com/TheR1D/shell_gpt/pull/557

**Comment by jeanlucthumm** [comment]
See https://github.com/TheR1D/shell_gpt/pull/557#issuecomment-2106036727
